IL2C.Core/ILConveters/ArithmeticalConverters.cs

```diff
@@ -310,4 +310,46 @@ internal sealed class ShiftLeftConverter : ShiftConverter
 
         public override ShiftDirection Direction => ShiftDirection.Left;
     }
+
+    internal sealed class ShiftRightUnConverter : ShiftConverter
+    {
+        public override OpCode OpCode => OpCodes.Shr_Un;
+
+        public override ShiftDirection Direction => ShiftDirection.Right;
+    }
+
+    internal sealed class NegConverter : InlineNoneConverter
```

Comment:
I wrote ckfinite(float/double) opcode implementation in C to conform with these tests https://github.com/dotnet/runtime/blob/f9a2449/src/tests/JIT/IL_Conformance/Old/Base/ckfinite.il#L52. 

Here is the C version: https://godbolt.org/z/vxMYrvb8d passing all conformance tests. coreclr uses the same approach by checking if all exponent bits are 1 to find out if float/double number is infinite or NaN, and emits ArithmeticException.

Was looking at the place to implement it in IL2C. Note: in this case, we need the raw single/double number in binary storage variable (e.g. uint32_t and uint64_t respectively) as it is represented in IL (IEEE 754 floating-point); without converting it to C float where it may or may not conform to IEEE 754 fp (ISO C99 is relaxed on the format...).

Feel free to implement ckfinite too, as this one is pretty simple. Otherwise, I'll take a stab at it; once this PR is merged. :slightly_smiling_face:

Reply:
Good news! I will include NaN handler and ckfinite implementation when after merged this PR. (Memoized: https://github.com/kekyo/IL2C/issues/112)